### ISSUE TITLE
Signature values mixed up using HardhatProvider

### ISSUE THREAD

**Issue by MrToph** [nomiclabs/hardhat#1175]
Hi,

I'm using a local hardhat network forking from ropsten and when I get [the following transaction](https://ropsten.etherscan.io/tx/0xabc467bedd1d17462fcc7942d0af7874d6f8bdefee2b299c9168a216d3ff0edb) using the ethersjs provider, the tx's `r` and `s` values are mixed up for me. I configured my hardhat network to use an alchemy endpoint.
This is my code:

```js
import hre from "hardhat";
const { ethers } = hre;
// correct r and s values are returned using this provider
// const provider = new ethers.providers.JsonRpcProvider(`https://eth-ropsten.alchemyapi.io/v2/SECRET`)

// wrong values are returned using the configured hardhat network forking ropsten
const provider = hre.ethers.provider
const firstTxHash = `0xabc467bedd1d17462fcc7942d0af7874d6f8bdefee2b299c9168a216d3ff0edb`;
const firstTx = await provider.getTransaction(firstTxHash);
console.log(`firstTx`, JSON.stringify(firstTx, null, 4));

// supposed to be these values, but r and s is interchanged
// r: '0xa5522718c0f95dde27f0827f55de836342ceda594d20458523dd71a539d52ad7',
// s: '0x5710e64311d481764b5ae8ca691b05d14054782c7d489f3511a7abf2f5078962',
```

Note that I get the correct r and s values when creating a new JsonRpcProvider using the Alchemy's node URL directly.
That's why I've been thinking that this might be an issue with hardhat.
This is the hardhat config I'm using:

```js
import dotenv from "dotenv";
dotenv.config(); // load env vars from .env
import { task, HardhatUserConfig } from "hardhat/config";
import "@nomiclabs/hardhat-waffle";

const { ARCHIVE_URL, MNEMONIC } = process.env;

const accounts = {
  mnemonic: MNEMONIC,
};

// Go to https://hardhat.org/config/ to learn more
const config: HardhatUserConfig = {
  solidity: {
    compilers: [
      { version: "0.7.3" }
    ],
  },
  networks: {
    ropsten: {
      url: ARCHIVE_URL,
      accounts,
    },
    hardhat: {
      accounts,
      forking: {
        url: ARCHIVE_URL, // https://eth-ropsten.alchemyapi.io/v2/SECRET`,
        blockNumber: 9389313,
      },
    },
  },
  mocha: {
    timeout: 300 * 1e3,
  }
};
```

Ref [ethersjs issue](https://github.com/ethers-io/ethers.js/issues/1224)

**Comment by scnale** [comment]
We recently merged a fix (#1158) for this and it'll be out in the next release.

**Comment by alcuadrado** [comment]
This has been fixed in v2.0.8. Thanks for reporting it!

https://github.com/nomiclabs/hardhat/releases/tag/hardhat-core-v2.0.8
